packages/wallet-sdk/src/util/encoding.ts

```diff
@@ -0,0 +1,56 @@
+import { hexToBytes, numberToHex, trim } from 'viem';
+
+export function base64ToBase64Url(base64: string): string {
+  return base64.replaceAll('+', '-').replaceAll('/', '_').replace(/=+$/, '');
+}
+
+export function arrayBufferToBase64Url(buffer: ArrayBuffer): string {
+  // First convert to regular base64
+  const base64String = btoa(String.fromCharCode(...new Uint8Array(buffer)));
+
+  // Then convert to base64url
+  return base64ToBase64Url(base64String);
+}
+
+export function convertCredentialToJSON(credential: any) {
```

Comment:
We should avoid type any. Either have a credential type or set it to unknown and assert the params

```tsx
export function convertCredentialToJSON(credential: unknow) {
  assertCredential(credential);
  // ... rest
}

type Credential = {
    id: string;
    rawId: string;
    response: {
      authenticatorData: AuthenticatorDat;
      clientDataJSON: ClientDataJSON;
      signature: Hex;
    },
    type: "" | "", // possible types
}

function assertCredential(credential: unknown): asserts is Credential {
  // validate and throw
}
```

Reply:
good callout, thanks. i moved the remaining preparation logic from the rpc method handler into here as well so now it takes the output from ox's `sign` and serializes it to JSON